lexer_test.go

```diff
@@ -232,6 +233,8 @@ func (s *testLexerSuite) TestIdentifier(c *C) {
 		{"1_x", "1_x"},
 		{"0_x", "0_x"},
 		{replacementString, replacementString},
+		{"9e", "9e"},
+		{"9eTSs", "9eTSs"},
```

Comment:
maybe need a test-case for `9e9e` :shipit:

Reply:
`9e9e` is not recognized as an single identifier though.

```sql
mysql> select 1 9e9e;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '9e9e' at line 1

mysql> select 9e9e;
+------------+
| e          |
+------------+
| 9000000000 |
+------------+
1 row in set (0.00 sec)
```

By extension, could we also check the following:

```sql
mysql> select 1 0xaaahhh;
+----------+
| 0xaaahhh |
+----------+
|        1 |
+----------+
1 row in set (0.00 sec)

mysql> select 1 0b123;
+-------+
| 0b123 |
+-------+
|     1 |
+-------+
1 row in set (0.00 sec)

mysql> select 1 0x3p4;
+-------+
| 0x3p4 |
+-------+
|     1 |
+-------+
1 row in set (0.00 sec)
```

Reply:
9e9e = 9e9 + e @lysu

Reply:
```
mysql> select 9e9e;
+------------+
| e          |
+------------+
| 9000000000 |
+------------+
1 row in set (0.00 sec)

mysql> select 9e9 e;
+------------+
| e          |
+------------+
| 9000000000 |
+------------+
1 row in set (0.00 sec)
```

@kennytm

Reply:
Test added @lysu

Reply:
```
mysql> create table t (0b123 int);
Query OK, 0 rows affected (0.19 sec)
```

You're right ... @kennytm 
0b123 is also an identifier!